src/dynarec/rv64/dynarec_rv64_f0.c

```diff
@@ -533,20 +550,43 @@ uintptr_t dynarec64_F0(dynarec_rv64_t* dyn, uintptr_t addr, uintptr_t ip, int ni
                         ed = xRAX+(nextop&7)+(rex.b<<3);
                         emit_add32c(dyn, ninst, rex, ed, i64, x3, x4, x5, x6);
                     } else {
+                        SMDMB();
                         addr = geted(dyn, addr, ninst, nextop, &wback, x2, x1, &fixedaddress, rex, LOCK_LOCK, 0, (opcode==0x81)?4:1);
                         if(opcode==0x81) i64 = F32S; else i64 = F8S;
+                        // TODO: Add support for BOX4_DYNAREC_ALIGNED_ATOMICS.
+                        ANDI(x1, wback, (1 << (rex.w + 2)) - 1);
+                        BNEZ_MARK3(x1);
+                        // Aligned
                         MARKLOCK;
                         LRxw(x1, wback, 1, 1);
-                        if(i64>=-2048 && i64<2048)
+                        if (i64 >= -2048 && i64 < 2048)
                             ADDIxw(x4, x1, i64);
                         else {
                             MOV64xw(x4, i64);
                             ADDxw(x4, x1, x4);
                         }
                         SCxw(x3, x4, wback, 1, 1);
                         BNEZ_MARKLOCK(x3);
-                        IFX(X_ALL|X_PEND)
+                        B_MARK_nocond;
+                        MARK3;
+                        // Unaligned
+                        ANDI(x5, wback, -(1 << (rex.w + 2)));
+                        MARK2; // Use MARK2 as a "MARKLOCK" since we're running out of marks.
+                        LDxw(x6, wback, 0);
+                        LRxw(x1, x5, 1, 1);
+                        if (i64 >= -2048 && i64 < 2048)
+                            ADDIxw(x4, x6, i64);
+                        else {
+                            MOV64xw(x4, i64);
```

Comment:
not really critical at all, but that MOV can probably be done out of the LR/SC loop.